common/src/main/java/io/confluent/connect/storage/common/ParentValueRecommender.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.connect.storage.common;
+
+import org.apache.kafka.common.config.ConfigDef;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A recommender which decides visibility based on the value of a parent config.
+ */
+public class ParentValueRecommender implements ConfigDef.Recommender {
+
+  protected String parentConfigName;
+  protected Object parentConfigValue;
+
+  public ParentValueRecommender(String parentConfigName, Object parentConfigValue) {
+    this.parentConfigName = parentConfigName;
+    this.parentConfigValue = parentConfigValue;
+  }
+
+  @Override
+  public List<Object> validValues(String name, Map<String, Object> connectorConfigs) {
+    return Collections.emptyList();
```

Comment:
Should the constructor also take a collection (or array for convenience) of recommended values that are valid when the parent value matches the specified value? Either way, we don't really want the list of valid values being empty, right?

Then, the different connectors could instantiate this recommender instance with:

```
private static final ParentValueRecommender AVRO_COMPRESSION_RECOMMENDER
      = new ParentValueRecommender(FORMAT_CLASS_CONFIG, 
                                   AvroFormat.class, 
                                   AVRO_SUPPORTED_CODECS);

Reply:
I am a bit hesitant to add a static collection here. IIUC, the validValues(..) method returns a list which is computed based on the properties (parsed so far). Ideally, we would want to pass in a lambda function or method reference.

Reply:
IIRC, `validValues(...)` is called when the configuration property is considered to be visible, and so the valid values for a visible AvroCodec property would always be the `AVRO_SUPPORTED_CODECS`. But we could enforce that as well by having `validValues(...)` call `visible(...)` and return an empty list if not visible and the supplied values when visible.

But I understand you're point. Unfortunately we're still on Java 7 and can't use method references or lambdas, which makes this harder. I'd say it's okay for this class to not be quite that flexible now since we don't need it. If somebody needs it in the future, they might have to override `validValues` or modify it to take a `Supplier<List<Object>>`. In the meantime, let's just be sure that the JavaDoc for the constructor is clear on how the supplied valid values would be used.

Reply:
Sounds good. Updated the Javadoc.